src/pk-engine.c

```
@@ -891,8 +896,22 @@ pk_engine_offline_upgrade_file_changed_cb (GFileMonitor *file_monitor,
 {
-	if (!g_network_monitor_get_network_available (network_monitor))
-		return PK_NETWORK_ENUM_OFFLINE;
+	if (!g_network_monitor_get_network_available (network_monitor)) {
+		g_autoptr(GError) error = NULL;
+		g_autoptr(GSocketConnectable) addr = g_network_address_new("1.2.3.4", 80);
+		syslog (LOG_DAEMON | LOG_NOTICE, "pk_engine_get_network_state: Gio::NetworkMonitor reports that network is offline");
+		if (g_network_monitor_can_reach(network_monitor, addr, NULL, &error)) {
+			syslog (LOG_DAEMON | LOG_NOTICE, "pk_engine_get_network_state: Gio::NetworkMonitor reports that network is offline, but also claims that the internet is reachable. Is NetworkManager configured correctly?");
+
+			/* this isn't exactly true, but it's what the UI expects */
+			if (g_network_monitor_get_network_metered (network_monitor))
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)